### ISSUE TITLE
websocketpp::exception::what() gives unexpected results

### ISSUE THREAD

**Issue by fabioang** [zaphoyd/websocketpp#397]
When `m_msg` is empty `what()` returns a pointer which is out of scope
```cpp
    virtual char const * what() const throw() {
        if (m_msg.empty()) {
            return m_code.message().c_str();   // temporary goes out of scope
        } else {
            return m_msg.c_str();
        }
    }
```
A possible solution could be:
```cpp
    virtual char const * what() const throw() {
        if (m_msg.empty()) {
            m_msg = m_code.message().c_str();
        }
        return m_msg.c_str();
    }
private:
    mutable std::string m_msg; // added mutable
    lib::error_code m_code;
```

Regards,
Fabio

**Comment by zaphoyd** [comment]
This should be fixed on the develop branch now

**Comment by zaphoyd** [comment]
The fix for this issue is now present in the 0.5.0 release on the master branch.
